packages/commonwealth/server/util/parseUserMentions.ts

```diff
@@ -1,4 +1,47 @@
-export const parseUserMentions = (text): any[] => {
+import { ServerError } from '@hicommonwealth/core';
+import { DB } from '@hicommonwealth/model';
+import { NotificationCategories, ProposalType } from '@hicommonwealth/shared';
+import { EmitOptions } from '../controllers/server_notifications_methods/emit';
+
+export type UserMention = {
+  profileId: string;
+  profileName: string;
+};
+
+export type UserMentionQuery = {
+  address: string;
+  user_id: number;
+}[];
+
+const hash = (mention: UserMention) =>
+  `name:${mention.profileName}id:${mention.profileId}`;
+
+// Will return unique mentions of the diff between current and previous mentions
+export const findMentionDiff = (
+  previousMentions: UserMention[],
+  currentMentions: UserMention[],
+) => {
+  const previousMentionHashes = new Set(previousMentions.map(hash));
+
+  return currentMentions.filter((mention) => {
+    const hashedMention = hash(mention);
+
+    // If it exists in previous hash set, it is not a new mention
+    if (previousMentionHashes.has(hashedMention)) {
+      return false;
+    }
+
+    // otherwise add it to the previous hash set so that we don't re-include duplicates
+    previousMentionHashes.add(hashedMention);
+
+    return true;
+  });
```

Comment:
Map/filter/reduce functions should generally be read-only since there's a bit more cognitive load if there's mutations inside of it. The lodash `differenceWith` could probably achieve the same result?

But I see there's a unit test for it so if it works, it works.